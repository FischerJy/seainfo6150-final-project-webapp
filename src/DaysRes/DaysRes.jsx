import React, { createRef, useEffect, useState } from "react";
import dayjs from "dayjs";
import { useParams } from "react-router";
import { Link } from "react-router-dom";

export default function Days() {
  const { lat, lon, name } = useParams();

  const [info, setInfo] = useState({
    daily: [
      {
        temp: {},
      },
    ],
  });

  useEffect(() => {
    var getlist = ({ lat, lon }) => {
      fetch(
        `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&exclude=current,minutely,hourly,alerts&appid=d2c0715f020c3a6cd3e893289a84cbc7&units=metric`
      )
        .then((res) => res.json())
        .then((res) => {
          console.log("res -> :", res);
          setInfo(res);
        });
    };
    getlist({ lat, lon });
    return () => {
      getlist = () => {};
    };
  }, [lat, lon]);

  return (
    <div className="main">
      <br />
      <br />
      <div>
        <h1>{name} Weathers Of 7 Days</h1>

        <div className="card-wrap">
          {info.daily.map((it, index) => (
            <div className="card" key={index}>
              <Link to={`/day/${name}/${lat}/${lon}/${it.dt}`}>
                <div>
                  <p>
                    Time:
                    {dayjs(it.dt * 1000).format("YYYY-MM-DD")}
                    <br />
                    {new Date(it.dt * 1000).toLocaleDateString("en", {
                      weekday: "long",
                    })}
                  </p>
                  <p>Temp:{it.temp.day}</p>
                </div>
              </Link>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}
