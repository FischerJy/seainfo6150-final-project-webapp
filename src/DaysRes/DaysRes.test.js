import React from "react";
import { render } from "@testing-library/react";
import DaysRes from "./DaysRes.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the DaysRes component correctly", () => {
    const { container } = render(<BrowserRouter>
        <DaysRes />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});