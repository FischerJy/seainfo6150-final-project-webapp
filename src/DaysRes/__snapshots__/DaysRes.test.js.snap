// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DaysRes component correctly 1`] = `
<div>
  <div
    class="main"
  >
    <br />
    <br />
    <div>
      <h1>
         Weathers Of 7 Days
      </h1>
      <div
        class="card-wrap"
      >
        <div
          class="card"
        >
          <a
            href="/day/undefined/undefined/undefined/undefined"
          >
            <div>
              <p>
                Time:
                Invalid Date
                <br />
                Invalid Date
              </p>
              <p>
                Temp:
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
